js/app/render-model/blogs.js

```diff
@@ -0,0 +1,15 @@
+import NodeArtist from "../../lib/node-artist/index.js";
+
+export default function({ blogs, paging }) {
```

Comment:
Ok can do. What would a meaningful name be here?

Reply:
In app.js (line 8) you are calling it  as `const preparingData = prepareData(pageOptions);`
So, `prepareData` ?

Could you also define the input parameter as `pageOptions `?

Reply:
You mean name the parameter rather than destructuring it?

Reply:
Yes, so it reads:

`export default function prepareData(pageOptions)`

Reply:
Sorry, not here, in index.js:

`export default async function prepareData(pageOptions)`

Reply:
To me the current version communicates _This is a function which you can call with any object having `blogs` and `paging` properties_

```
export default function({ blogs, paging }) {

    if (!blogs) return;
```

To me the suggested change communicates _This is a function named renderBlogs which you can call with a pageOptions object_
```
export default function renderBlogs(pageOptions) {

    const { blogs, paging } = pageOptions
    if (!blogs) return;
```

Reply:
I guess I'm not sure if defining the extra variable adds to or removes clarity. Is it partly because about the destructuring syntax is unfamiliar?

Reply:
May well be!

Reply:
But I use that `blogs` variable within the function further down. Are you saying in general to prefer accessing properties of objects instead of pulling them out into variables?

Reply:
I think you are proposing this:
```
import NodeArtist from "../../lib/node-artist/index.js";

export default function renderBlogs(pageOptions) {

    if (!pageOptions.blogs) return;

    const blogListItemArtist = NodeArtist("#blog-list-item");

    const blogList = document.querySelector("ol.blogs");
    for(let blog of pageOptions.blogs) {
        const listItem = blogListItemArtist(blog);
        blogList.appendChild(listItem);
    }

}
```

Reply:
Can you explain why this feels better to you? I'm genuinely curious about the ability of the code to communicate intent here. My original version was trying to:
1. make it very clear at the top of the function exactly what was expected of the caller (blogs and posts)
    - counter argument is that I immediately check that blogs is not null anyway, so it is fairly obvious in any case
1. prevent repeated property access as it makes the computer work harder 
    - counter argument is that the JIT compiler probably does that for us in the background anyway
1. try to make the code as terse as possible (but without making it harder to read)
    - counter argument is that it seems to have the opposite effect in this instance?

Reply:
It's my fault for commenting in the wrong place, but essentially yes, with a slight difference for this particular example:

```
import NodeArtist from "../../lib/node-artist/index.js";

export default function renderBlogs(viewModel) {

    if (!viewModel.blogs) return;

    const blogListItemArtist = NodeArtist("#blog-list-item");

    const blogList = document.querySelector("ol.blogs");
    for(let blog of viewModel.blogs) {
        const listItem = blogListItemArtist(blog);
        blogList.appendChild(listItem);
    }

}
```

Reply:
> Can you explain why this feels better to you? I'm genuinely curious about the ability of the code to communicate intent here. My original version was trying to:

It may be down to personal preference (and experience?) in the end, but I when I look at the code as a whole, render function in index.js accepts a model so I expect a model to be passed to it when referenced. Also, if we extend this model in future we wouldn't then need to amend the way it is referenced by having to add new properties to our calls?

Admittedly, viewModel should be renamed to something like blogsViewModel for better clarity.

Reply:
`render function in index.js accepts a model so I expect a model to be passed to it when referenced` - how does the destructuring syntax fail to communicate that to you?

Reply:
> `render function in index.js accepts a model so I expect a model to be passed to it when referenced` - how does the destructuring syntax fail to communicate that to you?

It is not a question of 'failing to communicate' - as I said, it may be down to personal preference.